src/test/ui/lint/force-warn/allowed-cli-deny-by-default-lint.rs

```diff
@@ -1,10 +1,9 @@
 // --force-warn $LINT causes $LINT (which is deny-by-default) to warn
 // despite $LINT being allowed on command line
-// compile-flags: -A const_err --force-warn const_err
+// compile-flags: -A arithmetic_overflow --force-warn arithmetic_overflow
 // check-pass
 
-const C: i32 = 1 / 0;
-//~^ WARN any use of this value will cause an error
-//~| WARN this was previously accepted by the compiler
-
-fn main() {}
+fn main() {
+    let _val = 1_i32 << 32;
+    //~^WARN: attempt to shift left by `32_i32`, which would overflow
+}
```

Comment:
@rylev @ehuss (who edited this file in the past) I don't have a clue what this is testing, but it has to switch from const_err to something else. (It's not a good idea to use a lint for such a test that is planned to be a hard error in the future. Just introduces more work for whoever is doing that transition eventually.)  From the comments it sounds like it needs to be a deny-by-default lint. `arithmetic_overflow` is a deny-by-default lint but here it does not behave the way the test wants it to. No warning is printed. I am at a loss.

Reply:
Oh, arithmetic_overflow does not trigger with `--emit metadata`. Wtf?

Is there any deny-by-default lint that also triggers for check builds?

Reply:
I found a suitable lint: mutable_transmutes